### DISCUSSION TITLE
BB Create Application POST API json payload?

### DISCUSSION THREAD

**Discussion by RSPont** [Budibase/budibase#5381] [Help]
is there a way to duplicate app layouts with the api? implement a json payload with specifications of an exported app?

**Comment by melohagan** [comment]
Hi @RSPont 

This isn't possible with the budibase api. Is there a particular reason you want to do it this way?

**Reply by RSPont** [reply]
My use case is more customer facing than internal, so when a user makes an account I want to assign them a new workspace with a premade layout

**Reply by melohagan** [reply]
So in this case, is each workspace a new app specific to each new user? 
This discussion may provide some more context if I've understood correctly: https://github.com/Budibase/budibase/discussions/3866

**Reply by RSPont** [reply]
Each worskpace is given per organization, which have their own users.
We wanted to duplicate layouts and give each org their own app to prevent "cross contamination"
We can add users to the app with the new api -great addition by the way-
The problem is that we can't automate the duplication. Sure, we can do it manually. Exporting and importing json config files, however it would be preferrable but I understand why this isn't a priority.
plus, there most probably is a way around this by querying couchdb.

**Reply by mjashanks** [reply]
@RSPont I believe that you are using multi-tenant to achieve this (I think we have chatted recently on a video call?).

Whilst we support multi-tenant, we don not currently have any APIs or Interfaces around it - we've only built the underlying backend code to enable it. 

Your suggestion here is a great one, and something that we should consider when it comes to developing a management interface around multi-tenancy, which is a bit further out for us - we simply have some more fundamental things to be working on.

That said, if multi-tenancy is highly sought after, and we have people paying us for it - we are likely to prioritize development around it.

For anyone else reading - the multi-tenancy feature will soon be pulled back into our enterprise plan (paid license) - so will not be open-source. It's only there at the moment because we use it ourselves, and hadn't developed the splitting of paid & open-source features.

**Reply by RSPont** [reply]
@mjashanks I think another thing that would work great for multitenancy is enviroment variables because I want to assign certain values but I don't really want to add another datasource just for a row of data. (which is also a security risk to expose something like that)

"I think we have chatted recently on a video call?"
You got me mixed up with someone else.

I think it's fair that you charge for multitenancy -just be mindful of the little guys- I get spooked just at looking at the contact sales prompt

**Reply by mjashanks** [reply]
@RSPont  ah cool - apologies! We will be releasing our pricing tiers in the next month or so!

Environment variable is also a great shout.
